app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -26,7 +26,7 @@ class AgendaRepositoryImpl(
     private val db: AgendaDatabase,
 ) : AgendaRepository {
 
-    override suspend fun getAgendaItems(localDate: LocalDate) = flow<Flow<List<AgendaItem>>> {
+    override suspend fun getAgendaItems(localDate: LocalDate) = flow<List<AgendaItem>> {
```

Comment:
There's no point in returning a flow and directly calling .first(). It will only return the first emission. The point of a flow is to be able to listen to changes at different times, so you also want to return the flow and not the result of it :)